### PR TITLE
Use conventional naming for example projects

### PR DESCRIPTION
**Pull request by simonjbeaumont**
### Motivation

The example packages were named using `CamelCase`, rather than `kebab-case` and also could use with some renaming to group them better and make it clear they are examples.

### Modifications

- Rename all the example Swift packages to use `kebab-case`.
- Include `-example` suffix in all example package names.
- Rename the Xcode project too to align with the Hello World examples.

### Result

Easier to browse the examples.

### Test Plan

CI.

### CODE REVIEW

File: Examples/curated-client-library-example/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Curated Library Client
+# Curated library client module
```
Comment (czechboy0) [inline Examples/curated-client-library-example/README.md:1]: Have we used the word "module" in any other example?

File: Examples/manual-generation-package-plugin-example/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# A command plugin invocation Client
+# Manual code generation using the Swift package plugin
```
Comment (czechboy0) [inline Examples/manual-generation-package-plugin-example/README.md:1]:
```suggestion
# Manual code generation using the Swift package command plugin
```

File: Examples/swagger-ui-endpoint-example/README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Swagger UI Endpoints Server
+# Server with Swagger UI Endpoint
```
Comment (czechboy0) [inline Examples/swagger-ui-endpoint-example/README.md:1]:
```suggestion
# Server with Swagger UI Endpoints
```

There are 3 endpoints, right? The raw file and the HTML page, plus the redirect.

**Review by czechboy0 — APPROVED** [review]
Just a few nits, otherwise looks great! We can update the Readme separately.

File: Examples/swagger-ui-endpoint-example/README.md (hunk 3 above)
Comment (simonjbeaumont) [reply Examples/swagger-ui-endpoint-example/README.md]: Went with "Server with OpenAPI document and Swagger UI endpoints" for now.

File: Examples/manual-generation-package-plugin-example/README.md (hunk 2 above)
Comment (simonjbeaumont) [reply Examples/manual-generation-package-plugin-example/README.md:1]: If it's using a manual workflow with a package plugin, it implies command plugin, right?

File: Examples/curated-client-library-example/README.md (hunk 1 above)
Comment (simonjbeaumont) [reply Examples/curated-client-library-example/README.md]: Updated to "Curated client library" for now.

**Comment by simonjbeaumont** [comment]
@czechboy0 OK, I updated the parent README and applied most of your suggestions. Given you approved already, I'll land this, and we can always follow up with more refinements before release.

**Pull request merged**